src/repositories/user.repository.ts

```diff
@@ -0,0 +1,52 @@
+import { v4 as uuidv4 } from "uuid";
+
+import { User } from "../models";
+
+export class UserRepository {
```

Comment:
I'm a little confused. Why do you use the repository name? Looks as service

Reply:
The idea of using a Repository pattern is to create simple CRUD operations with data, to be able to change one DB to another (or in-memory data to DB). 
I think the service is responsible for more complicated logic (e.g. getAutoSuggestUsers) and for combining data from different repositories (for example, users and departments).

Reply:
OK. But not needed here. When you will work with data base you will create DB Models